mirage/fixtures/addons.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable max-len */
+export default [
+    {
+        links: {},
+        url: 'http://www.box.com',
```

Comment:
This is fine, but generally it's better to use factories rather than fixtures. That will make it easier to have a specific number of items and controllable parameters. for tests. Not a big deal for this endpoint, because it's not going to change a lot and we don't do fancy things like pagination, but it's a good habit to get into. We usually only use fixtures if there's a really good reason to (for the above reasons and because fixtures are a bit more work to use).

Reply:
Yea reading through the mirage docs they basically said to use factories over fixtures - but in this case its static data (not really dynamic as it will always remain fixed) so a fixture was, apparently, correct for this specific endpoint.

Reply:
I agree we should use a factory and generate fake addon providers that are straight oauth (the most common case). But we can use fixtures for the providers that require form input (S3, Dataverse, Gitlab, and ownCloud) so that we can test the one-off components for those providers. Alternately, we could use traits to set up those providers in a factory. We briefly discussed dynamic forms defined in JSON schema (that would be a property in the API), but I think it's overkill here as it won't change often.